enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+
+
+### API Extensions & Controllers
+
+```mermaid
+flowchart TB
+
+subgraph HostedControlPlanes ETCD Backup API
+  HCSpec(hostedcluster.spec.etcd.backup) --> BackupSpec(*configv1alpha1.BackupSpec)
+  HCSpec --> StoreEndpoint(*hyperv1.storeEndpoint)
+
+  subgraph Original ETCD Backup API
+	  BackupSpec(*configv1alpha1.BackupSpec) --> EtcdBackupSpec(*configv1alpha1.EtcdBackupSpec)
+	  EtcdBackupSpec --> Schedule("Schedule --string--")
+	  EtcdBackupSpec --> TimeZone("TimeZone --string--")
+	  EtcdBackupSpec --> RetentionPolicy(RetentionPolicy)
+	  EtcdBackupSpec --> PVCName("PVCName --string--")
+	  RetentionPolicy(RetentionPolicy) --> RetentionType(RetentionType)
+	  RetentionType(RetentionType) --> RetentionNumber(RetentionNumber)
+	  RetentionNumber(RetentionNumber) --> MaxNumberOfBackups("MaxNumberOfBackups --int--")
+	  RetentionType(RetentionType) --> RetentionSize(RetentionSize)
+	  RetentionSize(RetentionSize) --> MaxSizeOfBackupsGb("MaxSizeOfBackupsGb --int--")
+  end
+
+  subgraph StorageEndpoint
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> BackupPrefix("BackupPrefix --string--")
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> Service("ServiceConfig --ServiceConfig--")
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> Credentials("Credentials --SecretRef--")
+
+    Service --> ServiceURL("ServiceURL --string--")
+    Service --> Region("Region --string--")
+    Service --> Tags("Tags --slice--")
+  end
+end
+```
+
+
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This is implemented directly for Hypershift.
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+**SNO:** The ETCD backup API and Controllers will work in the same way having 1-N nodes.
+**Microshift:** N/A
+
+### Implementation Details/Notes/Constraints
+
+In this section we will dissect the behaviour of the different controller to be implemented.
+
+#### ETCDBackupController
+
+- The `ETCDBackupController` initiates with a series of validations:
+  - Checks if the backup has already been performed by examining the HC and HCP labels (further details provided at the end of the process).
+  - Validates access permissions to the PVC and available free space. Errors in either could result in log entries, a new status.condition for the HC and HCP, and an event.
+  - Verifies the existence and health status of the HostedCluster necessary for performing the backup, as well as the presence of the PVC.
+  - If the PVC does not exist, the controller creates it and reschedules the backup for the next iteration, awaiting the status of Bound. If binding fails after a certain period, the controller will abandon the backup attempt and mark it as a failure.
+  - If the PVC already exists, the controller checks for another backup with the same final name. If found, it completes the process and updates the status.conditions of the HC and HCP accordingly.
+  - If the backup does not exist, the backup procedure proceeds.
```

Comment:
I'm not completely clear on how this PVC is being used. Looking at the flow here, it appears that there's a 1:many relationship between PVCs and Backups. meaning PVC is used over and over for backups.

if that's the case, why not make a 1:1 relationship to PVCs and backups, and use the labels/annotations on the PVC to help guide the retention policies for the backup within the cluster.

for example, every backup would result in a new PVC, we'd write an annotation on the PVC outlining when it expired, then garbage collection can be handled at the k8s resource level.

Reply:
Makes sense. This will save some computing resources. I've modified the enhancement in that way. Thanks